service/history/workflow/mutable_state_impl.go

```diff
@@ -4416,10 +4418,13 @@ func (ms *MutableStateImpl) ApplyWorkflowExecutionOptionsUpdatedEvent(event *his
 			ms.GetExecutionInfo().VersioningInfo = &workflowpb.WorkflowExecutionVersioningInfo{}
 		}
 		ms.GetExecutionInfo().VersioningInfo.VersioningOverride = &workflowpb.VersioningOverride{
-			Behavior: override.GetBehavior(),
+			Behavior:      override.GetBehavior(),
+			PinnedVersion: override.GetPinnedVersion(),
+		}
+		if d := override.GetDeployment(); d != nil {
```

Comment:
```suggestion
		if d := override.GetDeployment(); d != nil { // if the old Deployment field was populated instead of PinnedVersion
```
This took me a second to understand, so a comment might help